Cognitive Services version 2017-04-18

felixwa opened the pull request:
Generates SDK against Cognitive Services version 2017-04-18 swagger
file.
Fixes failed test cases accrodingly and adds 2 more test cases.
Bump version to 1.0.0.0 as Cognitive Services has been GA.

<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
<!--
Please add an informative description that covers that changes made by the pull request.

If you are regenerating your SDK based off of a new swagger spec, please add the link to the corresponding swagger spec pull request that has been merged in the azure-rest-api-specs repository
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

felixwa committed:
Cognitive Services version 2017-04-18

Generates SDK against Cognitive Services version 2017-04-18 swagger
file.
Fixes failed test cases accrodingly and adds 2 more test cases.
Bump version to 1.0.0.0 as Cognitive Services has been GA.

felixwa committed:
Cognitive Services version 2017-04-18

Generates SDK against Cognitive Services version 2017-04-18 swagger
file.
Fixes failed test cases accrodingly and adds 2 more test cases.
Bump version to 1.0.0.0 as Cognitive Services has been GA.

felixwa committed:
Bump VersionPrefix in csproj to 1.0.0

Bump VersionPrefix in csproj to 1.0.0

felixwa committed:
Bump VersionPrefix in csproj to 1.0.0

Bump VersionPrefix in csproj to 1.0.0

felixwa commented:
Generated based on Cognitive Services swagger file 2017-04-18: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-cognitiveservices/2017-04-18/swagger/cognitiveservices.json
Related PRs: 
https://github.com/Azure/azure-rest-api-specs/pull/1090
https://github.com/Azure/azure-rest-api-specs/pull/1149
https://github.com/Azure/azure-rest-api-specs/pull/1224

shahabhijeet commented on src/SDKs/CognitiveServices/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj:
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <Description>Microsoft Azure Management Cognitive Services Library</Description>
-    <VersionPrefix>0.3.0-preview</VersionPrefix>
+    <VersionPrefix>1.0.0</VersionPrefix>
```
@felixwa please replace the entire csproj with the following

<Project Sdk="Microsoft.NET.Sdk">
  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
  <PropertyGroup>
	<PackageId>Microsoft.Azure.Management.CognitiveServices</PackageId>
    <Description>Microsoft Azure Management Cognitive Services Library</Description>
    <VersionPrefix>1.0.0</VersionPrefix>
    <AssemblyName>Microsoft.Azure.Management.CognitiveServices</AssemblyName>
    <PackageTags>Microsoft Azure Cognitive Services management;Cognitive Services;Cognitive Services management;REST HTTP client;windowsazureofficial;netcore451511</PackageTags>
  </PropertyGroup>
  <PropertyGroup>
    <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
  </PropertyGroup>
</Project>

If you want to add ReadMe.txt please add .nuspec file, here is an example
https://docs.microsoft.com/en-us/nuget/schema/nuspec#including-content-files

felixwa replied:
Updated accordingly.
I will add that additional notice in another PR.

felixwa committed:
Cleans up csproj

Cleans up .csproj file as shahabhijeet suggested.

felixwa committed:
Cleans up csproj

Cleans up .csproj file as shahabhijeet suggested.

felixwa committed:
Fix format of csproj

Fix format of csproj

felixwa committed:
Fix format of csproj

Fix format of csproj

felixwa committed:
Update NuGet pacakges in test project to the latest.

Update NuGet pacakges in test project to the latest.

felixwa committed:
Update NuGet pacakges in test project to the latest.

Update NuGet pacakges in test project to the latest.

felixwa commented on src/SDKs/CognitiveServices/CognitiveServices.Tests/Tests/CognitiveServicesAccountTests.cs:
```diff
@@ -156,14 +155,14 @@ public void CognitiveServicesAccountListByResourceGroupTest()
                 var rgname = CognitiveServicesManagementTestUtilities.CreateResourceGroup(resourcesClient);
 
-                var accounts = cognitiveServicesMgmtClient.CognitiveServicesAccounts.ListByResourceGroup(rgname);
+                var accounts = cognitiveServicesMgmtClient.Accounts.ListByResourceGroup(rgname);
```
@shahabhijeet I have updated all nuget packages for test project though for ResourceManager (Microsoft.Azure.ResourceManager) I think we have used the latest (1.1.0-preview).  I don't see update option for it.

markcowl replied:
@felixwa the latest version is here: https://www.nuget.org/packages/Microsoft.Azure.Management.ResourceManager/1.6.0-preview

felixwa replied:
Okay, so actually it is a NEW NuGet package. Added.

shahabhijeet commented on src/SDKs/CognitiveServices/CognitiveServices.Tests/CognitiveServices.Tests.csproj:
```diff
@@ -27,4 +27,13 @@
     <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
   </ItemGroup>
+
+  <ItemGroup>
```
@felixwa any reason these references are added?
All these references are already implicitly added to your project, you do not need them again.
Please remove, or let me know the reason you made this change?

shahabhijeet replied:
@felixwa in order to use ResourceManagement, you will have to make that change inside CognitiveServicesManagementTestUtilities.cs, also please remove Compiler Constant #DNX451 from your test/SDK project.
The only constant you should be using is
#if FullNetFx (for Desktop framework version e.g. .NET 4.5.2 .NET 4.6 etc) and if you are targeting .NET Standard, please use
#if !FullNetFx

felixwa replied:
@shahabhijeet,
1. Removed those references. They were automatically added when I updated them in NuGet manager.
2. Add Microsoft.Azure.Management.ResourceManager 1.6.0-preview reference and changed UT code a little bit accordingly.
3. Re-recorded all unit tests.
4. Removed that #DNX451 constant.

shahabhijeet reviewed (changes requested):
Once you are done making suggested changes, remove "needs-revision" flag and add "Need review" flag

felixwa committed:
Remove DNX451 symbol

Remove DNX451 symbol

felixwa committed:
Merge branch 'Version2017-04-18' of https://github.com/felixwa/azure-sdk-for-net into Version2017-04-18

felixwa committed:
Adds ref to Microsoft.Azure.Management.ResourceManager 1.6.0-preview

1. Adds ref to Microsoft.Azure.Management.ResourceManager 1.6.0-preview
2. Adjust some test code accordingly.
3. Re-record all unit tests.
